### PR TITLE
refactor: rename `BundleStateWithReceipts` to `BlockExecutionOutcome`

### LINKED ISSUE #8727: Rename `BundleStateWithReceipts`
Think time to rename this to `BundleState` and have `with_receipts` and `with_requests` methods?

_Originally posted by @gakonst in https://github.com/paradigmxyz/reth/pull/8709#pullrequestreview-2108311231_

### PR DESCRIPTION
**Pull request by tcoratger**
This PR:
- Renames `BundleStateWithReceipts` to `BlockExecutionOutcome`
- Adds `with_receipts` and `with_requests`

Should close #8727.

### CODE REVIEW

File: crates/blockchain-tree/src/bundle.rs [hunk 1]
```diff
@@ -18,7 +18,7 @@ pub struct BundleStateDataRef<'a> {
 }
 
 impl<'a> BundleStateDataProvider for BundleStateDataRef<'a> {
-    fn state(&self) -> &BundleStateWithReceipts {
+    fn state(&self) -> &BlockExecutionOutcome {
```
Comment (mattsse) [inline crates/blockchain-tree/src/bundle.rs:21]: hmm, while naming is an improvement, this does not fit if we keep using it like this.

**Review by mattsse — COMMENTED** [review]
hmm, this is slightly more nuanced because a ton of variables still refer to this as `state` and other types also refer to this as state.

I very much prefer this new name because it's accurate.

but this clashes with how we refer to it, referring to this as state is even more confusing...

not sure how to proceed...

File: crates/blockchain-tree/src/bundle.rs (hunk 1 above)
Comment (tcoratger) [reply crates/blockchain-tree/src/bundle.rs]: Yes I thought we were saving it for another PR as there were a lot of changes to make but it's true that it doesn't make much sense to merge like that.

So I tried changing the names around `BlockExecutionOutcome` for overall consistency, hope I didn't forget any.

File: crates/storage/storage-api/src/state.rs [hunk 2]
```diff
@@ -242,12 +246,19 @@ pub trait BundleStateForkProvider {
...
 /// * block hashes of pending chain and canonical blocks.
 /// * canonical fork, the block on what pending chain was forked from.
-pub trait FullBundleStateDataProvider: BundleStateDataProvider + BundleStateForkProvider {}
+pub trait FullBundleStateDataProvider:
+    BlockExecutionDataProvider + BlockExecutionForkProvider
+{
+}
 
-impl<T> FullBundleStateDataProvider for T where T: BundleStateDataProvider + BundleStateForkProvider {}
+impl<T> FullBundleStateDataProvider for T where
```
Comment (mattsse) [inline crates/storage/storage-api/src/state.rs:261]: we should rename this as well

**Review by mattsse — CHANGES_REQUESTED** [review]
I like this a lot more than Bundlestate,

after looking at this, I would even consider dropping the `Block` prefix and just do `ExecutionOutput` this way it's not tied to `Block` because it could this can represent multiple blocks.

but I'd like more feedback here @Rjected @gakonst @shekhirin @rkrasiuk @rakita

File: crates/storage/storage-api/src/state.rs (hunk 2 above)
Comment (tcoratger) [reply crates/storage/storage-api/src/state.rs]: Done

**Comment by tcoratger** [comment]
> I like this a lot more than Bundlestate,
> 
> after looking at this, I would even consider dropping the `Block` prefix and just do `ExecutionOutput` this way it's not tied to `Block` because it could this can represent multiple blocks.
> 
> but I'd like more feedback here @Rjected @gakonst @shekhirin @rkrasiuk @rakita

@mattsse 
- On the one hand it will greatly alleviate the writing of the naming so it's not bad.
- But on the other hand `ExecutionResult` seems very generic and it may lack information about the interior of the object, right?

**Comment by shekhirin** [comment]
I like `ExecutionOutput`, it also makes names shorter everywhere, which is especially good for types like `FullBlockExecutionDataProvider`

**Comment by gakonst** [comment]
`ExecutionOutput` 👍

**Comment by mattsse** [comment]
ExecutionOutput it is then

**Comment by mattsse** [comment]
actually, can we roll this back to `Outcome` 1bd67d36d87446c5a35272b70a91124b302756cb, I like this more than output.
sorry about this

**Review by mattsse — APPROVED** [review]
ty 🫡

**Pull request merged**